src/main/scala/strawman/collection/Factories.scala

```diff
@@ -49,22 +49,26 @@ trait SpecificIterableFactory[-A, +C] extends FromSpecificIterable[A, C] {
 }
 
 /** Factory methods for collections of kind `* −> * -> *` */
-trait MapFactory[+CC[X, Y] <: Map[X, Y] with MapOps[X, Y, CC, _]] {
+trait MapFactory[+CC[X, Y]] {
 
   def empty[K, V]: CC[K, V]
```

Comment:
I think we really have to be careful not to introduce such cycles. Currently `fromIterable` implementations often rely on `empty`. Also, very often for concrete collection types we want to override this `empty` factory method for performance. So, I’m not sure it is helpful to have such a default implementation.